### DISCUSSION TITLE
Why do you have to open transactions when using jooq in micronaut, even if it is a simple select

### DISCUSSION THREAD

**Discussion by chengna106** [micronaut-projects/micronaut-data#1149] [Q&A]
Why do you have to open transactions when using jooq in micronaut, even if it is a simple select

**Comment by graemerocher** [comment]
Well every database interaction runs in a transaction, if there is no surrounding transaction you are in auto-commit mode

**Comment by verokarhu** [comment]
Even auto-commit opens a transaction AFAIK.

And doing multiple reads inside of a transaction is good practice. Otherwise you might end up updating your dataset in between selects from another source.
